kernel/kernel-api/src/test/scala/io/delta/kernel/internal/tablefeatures/TableFeaturesSuite.scala

```diff
@@ -560,13 +563,13 @@ class TableFeaturesSuite extends AnyFunSuite {
   }
 
   Seq("variantType", "variantType-preview", "variantShredding-preview").foreach { feature =>
-    checkWriteUnsupported(
+    checkWriteSupported(
       s"validateKernelCanWriteToTable: protocol 7 with $feature, " +
         s"metadata doesn't contains $feature",
       new Protocol(3, 7, singleton(feature), singleton(feature)),
       testMetadata())
 
-    checkWriteUnsupported(
+    checkWriteSupported(
```

Comment:
I think this test metadata needs to include variant right?

Reply:
This one already does (line 576), but it looks like the one above doesn't, I'll add that good catch!

Reply:
oh actually the intent of the test above was to validate in case the feature was not contained in the metadata, so i think it was good as is actually

Reply:
`includeInvariant` includes invariants in the metadata AFAIK, not _variants_ haha

Reply:
but yes the above one should not have variant in the metadata, and this one should